src/csg/csg.jl

```diff
@@ -0,0 +1,64 @@
+"""
+Represents a context-sensitive grammar constraint.
+Implementations can be found in the Herb-AI/Constraints.jl repository.
+"""
+abstract type Constraint end
+
+"""
+Structure representing context-sensitive grammar
+Extends ExprRules.Grammar with constraints
+"""
+struct ContextSensitiveGrammar <: Grammar
```

Comment:
I guess this PR came before probabilities?
So there will be no conflict, I guess.

Reply:
Indeed, I merged this branch in the branch for probabilistic grammars so I could continue working on that before waiting for this to be merged.